### DISCUSSION TITLE
Can't disable GitHub Pages for this repository

### DISCUSSION THREAD

**Discussion by DavidSchirduan** [community/community#22018] [Other Feature Feedback, Questions, & Ideas]
<p>I recently moved to Netlify for deploying my jekyll site. I tried to disable my github pages site using <a href="https://docs.github.com/en/github/working-with-github-pages/unpublishing-a-github-pages-site#unpublishing-a-user-or-organization-site" rel="nofollow noopener">these instructions</a>, but I see this error when I select the “none” branch:</p>
<blockquote>
<p>You can’t disable GitHub Pages for this repository. Read more on how to <a href="https://docs.github.com/github/working-with-github-pages/unpublishing-a-github-pages-site#unpublishing-a-user-or-organization-site" rel="nofollow noopener">unpublish your GitHub Pages site</a>.</p>
</blockquote>
<p>The offered link just leads back to the same instructions. How can I unpublish my github pages site? Thanks!</p>

**Comment by colemanm** [comment]
<p>I’m having this same issue. In my case the repository name is still in the <code>username.github.com</code> format that used to be part of the GH Pages configuration (I think legacy), using the <code>master</code> branch. All of that still works fine to deploy to Netlify.</p>
<p>Does it require renaming the repository? Not sure <a class="mention" href="/davidschirduan">@DavidSchirduan</a> if yours is set up like this, as well…</p>

**Comment by DavidSchirduan** [comment]
<p>Yup, mine’s setup the same way. I tried creating and deleting branches, but github pages just grabs whatever is set as the default branch. I don’t want to delete the repo, but I will if I have to.</p>
<p>It’s not great that github FORCIBLY publishes my site now.</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/davidschirduan">@DavidSchirduan</a> and <a class="mention" href="/colemanm">@colemanm</a>, welcome to the GitHub Support Community!</p>
<p>GitHub Pages sites served from repositories that use the special naming scheme <code>username.github.io</code> can’t be unpublished as it’s expected that a Pages site will be served from these repos. In order to disable Pages for these repositories you’ll need to first rename it to something else.</p>
<p>I’ll pass this on to the documentation team though, there should be something on this page that explains why you aren’t able to disable Pages for these specific repos on that help page.</p>

**Reply by Anas-Dew** [reply]
You mean a github site can never be deleted?

**Reply by DocGreenRob** [reply]
that sucks, i am investigating some other issue and literally just clicked the freaking button and now everything is messed up. i was following bad chatgpt advise trying to change the default branch. 

now i got this problem too, i simply want to undo what  i just did... now in deeper hole

**Comment by colemanm** [comment]
<p>Thanks <a class="mention" href="/thomasshaped">@thomasshaped</a>!</p>
<p>I thought it might be this but didn’t want to touch the names unless it would work. I’ll rename with something else and see if I can get it working. :+1:</p>

**Comment by DavidSchirduan** [comment]
<blockquote>
<p>GitHub Pages sites served from repositories that use the special naming scheme <code>username.github.io</code> can’t be unpublished as it’s expected that a Pages site will be served from these repos.</p>
</blockquote>
<p>After I changed the repo name I was able to switch the github pages branch to “None” which seems to have disabled the build process.</p>
<p>Thank you!</p>

**Reply by tekknolagi** [reply]
This worked for me!

**Comment by echessa** [comment]
<p>The <a href="https://docs.github.com/en/enterprise/2.17/user/github/working-with-github-pages/unpublishing-a-github-pages-site" rel="nofollow noopener">old documentation</a> said that to unpublish a GitHub Pages site, you had to delete the branch that was the publishing source of the Pages site. If using a user site and publishing from the master branch, you first had to set another branch as the default before deleting master. (I’ve linked to an old version of GitHub Enterprise documentation because I couldn’t find an old version of Github documentation, but the instructions were similar).</p>
<p>I tried that and it worked. The site was unpublished and I didn’t have to rename the repo. I don’t know if this will be disabled in the future (so you really can’t unpublish a Pages site unless you delete or rename the repo), or if it’s the documentation that doesn’t cover it, but I thought I should leave a comment since you mentioned that you’ll notify the documentation team to make some changes.</p>

**Comment by AndyObtiva** [comment]
<p>This is definitely a bug because when I got to Settings → Pages, it says “GitHub Pages is currently disabled.”</p>
<p>But, when I visit the home page of my project (AndyObtiva/glimmer), it says “github-pages Active”, which is clearly a contradiction. And, the irony is if I follow the link (<a href="https://github.com/AndyObtiva/glimmer/deployments/activity_log?environment=github-pages" class="inline-onebox" rel="noopener nofollow ugc">Deployments · AndyObtiva/glimmer · GitHub</a>), and click “View Deployment”, it gives me a 404 error, thus affirming that it is disabled.</p>
<p>The only issue is that the home page of my project is showing “github-pages Active” when Settings/Deployment of my GitHub project show the GitHub Pages to be disabled. The project home page should now show github-pages at all given that it’s disabled or at least show it as Inactive.</p>
<p>We shouldn’t have to rename our repository to fix this. You can call it a missing feature if you prefer not to think of it as a bug, but it is definitely a bug from a consumer’s point of view.</p>
<p>Thank you for providing GitHub as a free open-source service in any case and I hope that you would fix this issue and keep GitHub alive! The competition does not even begin to touch it!</p>
<p>Cheers!</p>

**Comment by th-schwarz** [comment]
<p>I had the same issue and I found out, you can delete/remove your github-pages in Settings → Environments.</p>

**Reply by DocGreenRob** [reply]
Thanks!

**Reply by Notare** [reply]
LOL, why this isn't in the docs is beyond me.
Thanks!

**Reply by dumont-nicolas** [reply]
Holy fuck that was hidden! I've been pulling my fucking beard off.
Over three years since OP and that shit is still a problem?!

**Comment by AndyObtiva** [comment]
<p>Good catch! Thank you… that fixed it!!</p>
